From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct sched_class
Patch-mainline: never, kabi

In 3.12.56, commit 69836a46b1c986665c514106b0137984cacb18dc (sched:
Replace post_schedule with a balance callback list), upstream commit
e3fca9e7cbfb72694a21c886fcdf9f059cfded9c removed post_schedule hook
from struct sched_class. It indeed changed the layout of the
structure.

But given this is only scheduler-internal structure, protect the
member the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/sched/sched.h |    3 +++
 1 file changed, 3 insertions(+)

--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -1045,6 +1045,9 @@ struct sched_class {
 	void (*migrate_task_rq)(struct task_struct *p, int next_cpu);
 
 	void (*pre_schedule) (struct rq *this_rq, struct task_struct *task);
+#ifdef __GENKSYMS__
+	void (*post_schedule) (struct rq *this_rq);
+#endif
 	void (*task_waking) (struct task_struct *task);
 	void (*task_woken) (struct rq *this_rq, struct task_struct *task);
 
